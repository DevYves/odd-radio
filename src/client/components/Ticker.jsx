import React, {Component} from 'react';

class Ticker extends Component {
  render(){
    if(!this.props.selectedStation.name) {
      return(
        <div></div>
      );
    } else if(!this.props.selectedStation.frequency) {
      return(
        <div>
          <div className="ticker-wrap">
            <div className="ticker">
              <div id="ticker-station" className="ticker__item"> {this.props.selectedStation.name} </div>
              <div id="ticker-city" className="ticker__item">Out of - {this.props.selectedStation.city}</div>
              <div id="ticker-dance" className="ticker__item">♪┏(・o･)┛♪┗ ( ･o･) ┓</div>
              <div id="ticker-brand" className="ticker__item">ODDRAD™</div>
              <div id="ticker-yeah" className="ticker__item">☜(ﾟヮﾟ☜)</div>
            </div>
          </div>
        </div>
      );
    } else {
      return(
        <div>
          <div className="ticker-wrap">
            <div className="ticker">
              <div id="ticker-station" className="ticker__item"> {this.props.selectedStation.name} </div>
              <div id="ticker-freq" className="ticker__item">@ {this.props.selectedStation.frequency}</div>
              <div id="ticker-city" className="ticker__item">Out of - {this.props.selectedStation.city}</div>
              <div id="ticker-dance" className="ticker__item">♪┏(・o･)┛♪┗ ( ･o･) ┓</div>
              <div id="ticker-brand" className="ticker__item">ODDRAD™</div>
              <div id="ticker-yeah" className="ticker__item">☜(ﾟヮﾟ☜)</div>
            </div>
          </div>
        </div>
      );
    }
  }
}

export default Ticker;