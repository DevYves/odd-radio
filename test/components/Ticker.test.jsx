import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

import Ticker from '../../src/client/components/Ticker';

describe('Ticker', () => {
  const props = {
    selectedStation: {
        id: '1',
        name: 'CFBU',
        audioFeed: 'www.google.ca',
        streamType: 'audio/mp3',
        frequency: '90.1 FM',
        city: 'Hamilton',
        province: 'ON'
    }
  }

  const subject = <Ticker { ...props }/>;

  it('matches snapshot', () => {
    expect(renderer.create(subject)).toMatchSnapshot();
  });
});

describe('Ticker Station With No Frequency', () => {
  const props = {
    selectedStation: {
        id: '1',
        name: 'CFBU',
        audioFeed: 'www.google.ca',
        streamType: 'audio/mp3',
        frequency: '',
        city: 'Hamilton',
        province: 'ON'
      }
    }

    const subject = <Ticker { ...props }/>;

    it('displays all station info except station frequency', () => {
      const wrapper = shallow(subject);

      expect(wrapper.find('#ticker-freq').length).toBe(0);
    });

});