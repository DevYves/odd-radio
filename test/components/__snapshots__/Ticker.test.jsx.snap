// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticker matches snapshot 1`] = `
<div>
  <div
    className="ticker-wrap"
  >
    <div
      className="ticker"
    >
      <div
        className="ticker__item"
        id="ticker-station"
      >
         
        CFBU
         
      </div>
      <div
        className="ticker__item"
        id="ticker-freq"
      >
        @ 
        90.1 FM
      </div>
      <div
        className="ticker__item"
        id="ticker-city"
      >
        Out of - 
        Hamilton
      </div>
      <div
        className="ticker__item"
        id="ticker-dance"
      >
        ♪┏(・o･)┛♪┗ ( ･o･) ┓
      </div>
      <div
        className="ticker__item"
        id="ticker-brand"
      >
        ODDRAD™
      </div>
      <div
        className="ticker__item"
        id="ticker-yeah"
      >
        ☜(ﾟヮﾟ☜)
      </div>
    </div>
  </div>
</div>
`;
